### PR TITLE
chore: Optimize Contracts

### PR DESCRIPTION
**Pull request by Sneh1999**
**What changed? Why?**
Optimizing Contracts based on feedback

**Notes to reviewers**
I still need to implement 712 but it would require some backend changes as well, so I will do it in another PR

**How has it been tested?**

### CODE REVIEW

**Review by Zizzamia — APPROVED** [review]
Nice

**Comment by Sneh1999** [comment]
> Nice

@Zizzamia  this PR is not completed yet :) Still need to do some other optimizations

File: README.md [hunk 1]
```diff
@@ -89,13 +89,13 @@ forge build
 #### Step 6: Deploy contracts to Base Sepolia
 
 Create a `.env` file using the `.env.example` file provided in your contracts folder and add your private key. Make sure to add a `0x` in front of your key to convert it to a hex string.
-
+Note: Get an API key from [here](https://basescan.org/) for Base Sepolia by creating an account
```
Comment (awilliams1-cb) [inline README.md:92]: Nit: minor wording suggestion
```suggestion
Note: In order to verify your smart contract on Basescan, you need an API key. You can get an API key from [here](https://basescan.org/) for Base Sepolia by creating an account.
```

File: README.md [hunk 2]
```diff
@@ -89,13 +89,13 @@ forge build
 #### Step 6: Deploy contracts to Base Sepolia
```
Comment (awilliams1-cb) [inline README.md:89]:
```suggestion
#### Step 6: Deploy and verify contracts to Base Sepolia
```

File: apps/build-onchain-apps/contracts/README.md [hunk 3]
```diff
@@ -122,12 +122,13 @@ forge fmt
 ### Deploy to Base Sepolia
```
Comment (awilliams1-cb) [inline apps/build-onchain-apps/contracts/README.md:122]: Same suggestions as the other README
```suggestion
### Deploy and verify contracts to Base Sepolia
```

File: apps/build-onchain-apps/contracts/README.md [hunk 4]
```diff
@@ -122,12 +122,13 @@ forge fmt
...
 
 Create a `.env` file using the `.env.example` file provided in your contracts folder and add your private key. Make sure to add a `0x` in front of your key to convert it to a hex.
+Note: Get an API key from [here](https://basescan.org/) for Base Sepolia by creating an account
```
Comment (awilliams1-cb) [inline apps/build-onchain-apps/contracts/README.md:125]:
```suggestion
Note: In order to verify your smart contract on Basescan, you need an API key. You can get an API key from [here](https://basescan.org/) for Base Sepolia by creating an account.
```

File: apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol [hunk 5]
```diff
@@ -23,13 +23,13 @@ contract SignatureMintERC721Test is Test {
...
     // --- Utility Methods ---
-    function signMessage(bytes32 message, uint256 key) internal returns (bytes memory) {
+    function signMessage(bytes32 message, uint256 key) internal pure returns (bytes memory) {
         (uint8 v, bytes32 r, bytes32 s) = vm.sign(key, message);
         bytes memory signature = toBytes(r, s, v);
         return signature;
     }
 
-    function genKeyPair() internal returns (address, uint256) {
+    function genKeyPair() view internal returns (address, uint256) {
```
Comment (awilliams1-cb) [inline apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol:32]: Nit: Solidity [style guide](https://docs.soliditylang.org/en/v0.8.23/style-guide.html) mentions visibility should go before mutability
```suggestion
    function genKeyPair() internal view returns (address, uint256) {
```

File: apps/build-onchain-apps/src/components/home/Guide.tsx [hunk 6]
```diff
@@ -24,10 +24,10 @@ forge test
 
 # Format
 forge fmt`;
-const codeStep6 = `# Create a .env file using the .env.example file provided in your contracts folder and add your private key.
-source .env
+const codeStep6 = `# Create a .env file using the .env.example file provided in your contracts folder and add your private key. Make sure to add a 0x in front of your key to convert it to a hex.
+Note: Get an API key from basescan.org for Base Sepolia by creating an account
```
Comment (awilliams1-cb) [inline apps/build-onchain-apps/src/components/home/Guide.tsx:28]:
```suggestion
Note: In order to verify your smart contract on Basescan, you need an API key. You can get an API key from [here](https://basescan.org/) for Base Sepolia by creating an account.
```

File: apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol [hunk 7]
```diff
@@ -64,9 +64,6 @@ contract SignatureMintERC721Test is Test {
         assertEq(signatureMintERC721.balanceOf(minter), 0);
         bytes32 messageToSign = signatureMintERC721.getBytesToSign(minter);
 
-        // Expect the FreeMint event to get called
-        vm.expectEmit(true, true, true, true);
-        emit SignatureMintERC721.Mint(minter, 1, 0);
```
Comment (awilliams1-cb) [inline apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol:69]: Question: why is this assertion being removed? Was it causing a problem?

**Review by awilliams1-cb — COMMENTED** [review]
Left some minor nits

File: apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol (hunk 7 above)
Comment (Sneh1999) [reply apps/build-onchain-apps/contracts/test/SignatureMintERC721.t.sol:69]: This emit event is called  here `signatureMintERC721.freeMint(minter, signature)` , we dont really need to test it seperatelt

File: README.md (hunk 2 above)
Comment (Sneh1999) [reply README.md:89]: done! thank you

File: README.md (hunk 1 above)
Comment (Sneh1999) [reply README.md:92]: done! thank you

**Comment by Sneh1999** [comment]
> Left some minor nits

Thanks Alexis! Pushed the changes based on the nits

**Review by awilliams1-cb — APPROVED** [review]

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**